spharpy/spherical.py

```diff
@@ -1005,3 +1007,413 @@ def sph_identity_matrix(n_max, type='n-nm'):
+        - ``'fuma'``: Follows the Furse-Malham (FuMa) convention.
+        (FuMa is only supported up to 3rd order)
+
+    - inverse_transform: Defines the type of inverse transform.
+     It can be 'pseudo_inverse', 'quadrature', or None.
+        - ``'pseudo_inverse'``: Uses the Moore-Penrose pseudo-inverse
+         for the inverse transform.
+        - ``'quadrature'``: Uses quadrature for the inverse transform.
+        - ``None``: No inverse transform is applied.
+
+    Parameters
+    ----------
+    n_max : int
+        Maximum spherical harmonic order
+    coordinates : :doc:`pf.Coordinates <pyfar:classes/pyfar.coordinates>` or
+        `sp.SamplingSphere <spharpy:classes/spharpy.samplings.coordinates>`
+        objects with sampling points for which the basis matrix is
+        calculated
+    basis_type : str, optional
+        Type of spherical harmonic basis, either ``'complex'`` or
+        ``'real'``. The default is ``'real'``.
+    normalization : str, optional
+        Normalization convention, either ``'n3d'``, ``'maxN'`` or
+         ``'sn3d'``. The default is ``'n3d'``.
+        (maxN is only supported up to 3rd order)
+    channel_convention : str, optional
+        Channel ordering convention, either ``'acn'`` or ``'fuma'``.
+        The default is ``'acn'``.
+        (FuMa is only supported up to 3rd order)
+    inverse_transform : str, optional
+        Inverse transform type, either ``'pseudo_inverse'`` or
+        ``'quadrature'``. The default is None.
+    phase_convention : string or None, optional
+        Whether to include the Condon-Shortley phase term.
+        The default is None.
+    """
+
+    def __init__(
+        self,
+        n_max,
+        coordinates,
+        basis_type="real",
+        normalization="n3d",
+        channel_convention="acn",
+        inverse_transform=None,
+        phase_convention=None
+    ):
+        # hidden attributes
+        self._n_max = None
+        self._coordinates = pf.Coordinates()
+        self._basis_type = None
+        self._weights = None
+        self._inverse_transform = None
+        self._channel_convention = None  # ch. ord. conv.
+        self._normalization = None  # gain norm. conv.
+        self._phase_convention = None
+
+        self.n_max = n_max
+        self.coordinates = coordinates
+        self.basis_type = basis_type
+        self.inverse_transform = inverse_transform
+        self.channel_convention = channel_convention
+        self.normalization = normalization
+        self.phase_convention = phase_convention
+
+        self._recompute_basis = True
+        self._recompute_basis_gradient = True
+        self._recompute_inverse = True
+        self._recompute_inverse_gradient = True
+
+        self._basis = None
+        (self._basis_gradient_theta,
+         self._basis_gradient_phi) = (None, None)
+        self._basis_inv = None
+        (self._basis_inv_gradient_theta,
+         self._basis_inv_gradient_phi) = (None, None)
```

Comment:
Do we need both? Instead of setting `recompute=True` we could set everything to `None` when changing parameters that require recompute.